src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestExtractFileServiceImpl.java

```diff
@@ -0,0 +1,172 @@
+            FileUtils.moveFile(new File(currentPathAndFile), new File(newPathFileNameAndExtension));
+        } catch (IOException e) {
+            LOG.error("Unable to move file From=" + currentPathAndFile + "=  To=" + newPathFileNameAndExtension, e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void setBatchInputFileService(BatchInputFileService batchInputFileService) {
+        this.batchInputFileService = batchInputFileService;
+    }
+
+    public BatchInputFileService getBatchInputFileService() {
+        return batchInputFileService;
+    }
+
+    public void setBatchInputFileType(BatchInputFileType batchInputFileType) {
+        this.batchInputFileType = batchInputFileType;
+    }
+
+    public BatchInputFileType getBatchInputFileType() {
+        return batchInputFileType;
+    }
+
+    public void setAcceptedDirectoryName(String acceptedDirectoryName) {
+        this.acceptedDirectoryName = acceptedDirectoryName;
+    }
+
+    public String getAcceptedDirectoryName() {
+        return this.acceptedDirectoryName;
+    }
+
+    public void setRejectedDirectoryName(String rejectedDirectoryName) {
+        this.rejectedDirectoryName = rejectedDirectoryName;
+    }
+
+    public String getRejectedDirectoryName() {
+        return this.rejectedDirectoryName;
+    }
+
+    public void setConcurRequestExtractFileValidationService(ConcurRequestExtractFileValidationService concurRequestExtractFileValidationService) {
+        this.concurRequestExtractFileValidationService = concurRequestExtractFileValidationService;
+    }
+
+    public ConcurRequestExtractFileValidationService getConcurRequestExtractFileValidationService() {
+        return this.concurRequestExtractFileValidationService;
+    }
+
+    private Object loadFile(String fileName) {
+        byte[] fileByteContent = safelyLoadFileBytes(fileName);
+        Object parsedObject;
+        parsedObject = getBatchInputFileService().parse(getBatchInputFileType(), fileByteContent);
+        return parsedObject;
+    }
+
+    private byte[] safelyLoadFileBytes(String fileName) {
+        InputStream fileContents;
+        byte[] fileByteContent;
+        try {
+            fileContents = new FileInputStream(fileName);
+        } catch (FileNotFoundException e1) {
+            LOG.error("Batch file not found [" + fileName + "]. " + e1.getMessage());
+            throw new RuntimeException("Batch File not found [" + fileName + "]. " + e1.getMessage());
+        }
+        try {
+            fileByteContent = IOUtils.toByteArray(fileContents);
+        } catch (IOException e1) {
+            LOG.error("IO Exception loading: [" + fileName + "]. " + e1.getMessage());
+            throw new RuntimeException("IO Exception loading: [" + fileName + "]. " + e1.getMessage());
+        } finally {
+            IOUtils.closeQuietly(fileContents);
+        }
+        return fileByteContent;
+    }
+
+    public void process(String fileName, Object parsedFileContents) {
```

Comment:
The FlatFileDataHandler is the class that contains the "processor" to deal with parsing each file into the POJO data map that the rest of this processing utilizes. The KFS server will generate a 404 error on startup if the file service interface does not extend it.